grails-app/controllers/uk/org/nbn/alerts/SavedSearchController.groovy

```diff
@@ -0,0 +1,236 @@
+     * eg by using annotation or however else the application enforces authentication
+     */
+    @RequireApiKey
+    def list() {
+        if (!params.userId) {
+            flash.errorMessage = "userId is a required parameter"
+            redirect(action: "index")
+            return
+        }
+
+        def user = User.findByUserId(params.userId)
+        if (!user) {
+            flash.errorMessage = "User not found"
+            redirect(action: "index")
+            return
+        }
+
+        def savedSearches = savedSearchService.getSavedSearch(params.userId)
+        render savedSearches as JSON
+    }
+
+    @RequireApiKey
+    def save() {
+        if (!params.userId) {
+            flash.errorMessage = "userId is a required parameter to save a search"
+            redirect(action: "mySavedSearches", params: [userId: params.userId])  // Add userId to redirect
+            return
+        }
+
+        def currentUser = User.findByUserId(params.userId)
+        if (!currentUser) {
+            flash.errorMessage = "User not found"
+            redirect(action: "mySavedSearches", params: [userId: params.userId])  // Add userId to redirect
+            return
+        }
+
+        try {
+            def savedSearch = savedSearchService.createSavedSearch(
+                params.userId,
+                params.name,
+                params.description,
+                params.searchRequestQueryUI
+            )
+
+            flash.message = "Search saved successfully"
+            redirect(action: "mySavedSearches", params: [userId: params.userId])  // This ensures we go back to My Saved Searches
+        } catch (Exception e) {
+            flash.errorMessage = "Error saving search: ${e.message}"
+            render(view: "create", model: [savedSearch: params])
+        }
+    }
+
+
+
+    @RequireApiKey
+    def updateApi() {
+        if (!params.id || !params.userId) {
+            response.status = HttpStatus.SC_BAD_REQUEST
+            render([error: "id and userId are required parameters"] as JSON)
+            return
+        }
+
+        def updatedSearch = savedSearchService.updateSavedSearch(params.id, params.userId, params.name, params.description, params.searchRequestQueryUI)
+        render([success: true, id: updatedSearch.id] as JSON)
+    }
+
+    @RequireApiKey
+    def delete() {
+        if (!params.id || !params.userId) {
+            response.status = HttpStatus.SC_BAD_REQUEST
+            render([error: "id and userId are required parameters", success: false] as JSON)
+            return
+        }
+
+        try {
+            savedSearchService.deleteSavedSearch(params.id, params.userId)
+            render([success: true, message: "Search deleted successfully"] as JSON)
+        } catch (Exception e) {
+            response.status = HttpStatus.SC_INTERNAL_SERVER_ERROR
+            render([error: "Failed to delete search", success: false] as JSON)
+        }
+    }
+
+ @RequireApiKey
+    def editApi() {
```

Comment:
great rid of editApi - only need one method. 

shouldn't it be update?

Reply:
deleted editApi method